test_suite/tests/test_de_error.rs

```diff
@@ -1454,6 +1454,14 @@ fn test_wrapping_overflow() {
     );
 }
 
+#[test]
+fn test_saturating_overflow() {
+    assert_de_tokens_error::<Saturating<u16>>(
+        &[Token::U32(65_536)],
+        "invalid value: integer `65536`, expected u16",
```

Comment:
I wonder whether this matches the behavior that most people would expect of this impl. Do you think we'll have users who expect this to saturate to 65535?

Reply:
Good question. I copied and adapted this test case from the `std::num::Wrapping` impl.

I think the point of the `Saturating` type is to treat very large numbers just like the maximum number, because one is not interested in how much bigger something is than the maximum. So yes, it is meaningful to make this a non-error. I'll update the PR tomorrow to make it tolerate exceedingly large numbers.

Reply:
Yes, that sounds all right to me. Thanks!

Reply:
I'm currently stuck at the following problem: Saturating can happen on two ends:

1) A value is lower than the allowed range
2) A value is higher than the allowed range

For example, if we want to create a `Saturating<u8>` from a `i8` or a `u16` we have:

`i8::MIN = -128` should map to `0u8`
`u16::MAX = 65535` should map to `255u8`

When I implement `Deserialize<'de> for Saturating<u8>` I have

```rust
match Deserialize::deserialize(deserializer) {
    Ok(number) => Ok(Saturating(number)),
    Err(e) => Ok(Saturating(u8::MAX)), // only handles values above the range correctly and disregards any other error conditions
}
```

The problem is that I do not know how to determine if we have a value below or above the range. The error `e` does not seem to provide any information on the what the original input was.